frontend/packages/kubevirt-plugin/src/selectors/config-map/sc-defaults.ts

```diff
@@ -16,31 +15,44 @@ const getSCConfigMapAttribute = (
     _.get(storageClassConfigMap, [
       'data',
       hasSubAttribute ? `${storageClassName}.${attributeName}` : attributeName,
-    ]) || defaultValue
+    ]) || null
   );
 };
 
 export const getDefaultSCVolumeMode = (
   storageClassConfigMap: ConfigMapKind,
   storageClassName: string,
-) =>
-  getSCConfigMapAttribute(
+) => {
+  const configMapDefault = getSCConfigMapAttribute(
     storageClassConfigMap,
     storageClassName,
     'volumeMode',
-    PVC_VOLUMEMODE_DEFAULT,
   );
 
+  const volumeMode = configMapDefault ? VolumeMode.fromString(configMapDefault) : null;
+
+  if (volumeMode) {
+    return volumeMode;
+  }
+
+  return storageClassName === 'local-sc' ? VolumeMode.FILESYSTEM : VolumeMode.BLOCK;
```

Comment:
why we should default to block on anything not 'local-sc', ( not a blocker ) , do we have a reason ?

Reply:
That comes from the design doc. We try to pull the defaults from the config map first. If nothing is found (the config map doesn't exist or there isn't an entry for the selected storage class), then these values are used as the fallbacks. Just to make sure, I'll check with the storage folks and make sure this logic works. If it doesn't, I'll fix it in a follow-up.